prov/rxm/src/rxm_domain.c

```diff
@@ -142,6 +183,7 @@ static void rxm_mr_init(struct rxm_mr *rxm_mr, void *context)
 	 * The key would be used in large message transfer protocol and RMA. */
 	rxm_mr->mr_fid.mem_desc = rxm_mr->msg_mr;
 	rxm_mr->mr_fid.key = fi_mr_key(rxm_mr->msg_mr);
+	rxm_mr->domain = domain;
```

Comment:
Thanks, I've updated the domain reference counting to be done at MR creation/close since the domain is referenced whether the MR is added to the reverse lookup map or not.